api/client/src/main/java/org/projectnessie/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -209,38 +208,16 @@ private static HttpServer createServer(int port, HttpHandler handler) {
       throw new UncheckedIOException(e);
     }
     server.createContext(CONTEXT_PATH, handler);
-    bind(server, port);
+    InetAddress local = InetAddress.getLoopbackAddress();
+    try {
+      server.bind(new InetSocketAddress(local, port), 0);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
     server.start();
     return server;
   }
 
-  private static void bind(HttpServer server, int port) {
```

Comment:
This was unnecessarily convoluted (not related to programmatic config in any way, but I seized the chance to fix it.)